packages/medusa/src/services/event-bus.ts

```diff
@@ -223,29 +237,47 @@ export default class EventBusService {
   async emit<T>(
     eventName: string,
     data: T,
-    options: Record<string, unknown> & EmitOptions = { attempts: 1 }
-  ): Promise<StagedJob | void> {
+    options?: Record<string, unknown> & EmitOptions
+  ): Promise<StagedJob | void>
+
+  async emit<
+    T,
+    TInput extends string | EmitData<T>[] = string,
+    TResult = TInput extends EmitData<T>[] ? StagedJob[] : StagedJob
+  >(
+    eventNameOrData: TInput,
+    data?: T,
+    options: Record<string, unknown> & EmitOptions = {}
+  ): Promise<TResult | void> {
     const globalEventOptions = this.config_?.projectConfig?.event_options ?? {}
 
+    const isBulkEmit = !isString(eventNameOrData)
+    const events: EmitData<T>[] = isBulkEmit
+      ? eventNameOrData
+      : [
+          {
+            eventName: eventNameOrData,
+            data: data!,
+            options,
```

Comment:
todo: I believe this should be `opts`: https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#queueaddBulk
```suggestion
            opts: options,
```